Remotion/Development/UnitTesting.Compilation/Roslyn/RoslynCompiler.cs

```diff
@@ -378,6 +377,70 @@ private void Compile(CompilerParameters options, string compilerFullPath, string
             }
         }
 
+        // Adapted version of Executor.ExecWaitWithCapture that works in Linux.
+        private static int ExecuteCommand(
+            string fileName,
+            string args,
+            string currentDir,
+            TempFileCollection tempFiles,
+            ref string outputName,
+            ref string errorName)
+        {
+            if (string.IsNullOrEmpty(outputName))
+              outputName = tempFiles.AddExtension("out");
+
+            if (string.IsNullOrEmpty(errorName))
+              errorName = tempFiles.AddExtension("err");
+
+            using (var outputWriter = new StreamWriter(CreateInheritedFile(outputName), Encoding.UTF8))
+            using (var errorWriter = new StreamWriter(CreateInheritedFile(errorName), Encoding.UTF8))
+            {
+                // Output the command line...
+                outputWriter.Write(currentDir);
+                outputWriter.Write("> ");
+                outputWriter.Write(fileName);
+                outputWriter.Write(" ");
+                outputWriter.WriteLine(args);
+                outputWriter.WriteLine();
+                outputWriter.WriteLine();
+
+                var psi = new ProcessStartInfo
+                          {
+                                  FileName = "dotnet",
+                                  Arguments = $"exec \"{fileName}\" {args}",
+                                  WorkingDirectory = currentDir,
+                                  RedirectStandardOutput = true,
+                                  RedirectStandardError = true
+                          };
+
+                using (var p = Process.Start(psi))
+                {
+                    p.OutputDataReceived += (s, e) =>
+                    {
+                        if (e.Data != null)
+                          outputWriter.WriteLine(e.Data);
+                    };
+                    p.ErrorDataReceived += (s, e) =>
+                    {
+                        if (e.Data != null)
+                          errorWriter.WriteLine(e.Data);
+                    };
+
+                    p.BeginOutputReadLine();
+                    p.BeginErrorReadLine();
+
+                    if (!p.WaitForExit(600_000))
+                      throw new ExternalException($"External command failed with exit code {p.ExitCode}: '{fileName} {args}'", 0x102);
+
+                    p.WaitForExit();
```

Comment:
why the second WaitForExit? Am I missing something in terms of logic? we wait for 600 seconds for the process to exit and then we wait again if there was no exception?

Also, why wait for 600 seconds?